### PR TITLE
fix: missing guarding leads to crashes under Android prior to API 23

### LINKED ISSUE #8702: [Android] Touch events broken in older Android versions
### Current behavior

After upgrading to Uno.UI 4.2.6 any click and tapped event will raise an "Java.Lang.NoSuchMethodError" related to motionevents, which will crash the entire app.

I detect the problem in my apps' telemetry and in the Android emulator for SDK 22 (Android 5.1)
The previous version 4.1.9 of Uno still worked well.

Did Uno introduce specific minimum versions for Android?

### Expected behavior

User can still click buttons or tap on elements. ;-)

### How to reproduce it (as minimally and precisely as possible)

Simply create a button and add a click event handler.

### Workaround

Downgrade to Uno 4.1.9

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

4.2.6

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by queequac**
GitHub Issue (If applicable): closes #8702

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
Pointer events lead to crashing apps under Android prior to SDK 23.

## What is the new behavior?

Works

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by carldebilly** [comment]
@queequac You'll need to follow Conventional Commits format.

**Comment by queequac** [comment]
Saw too late that I added the category to the PR, not the commit.
Renamed it and forced a push... good enough now?

**Review by MartinZikmund — APPROVED** [review]

**Comment by dr1rrb** [comment]
Thanks for your commit, unfortunately this is no longer enough as we are now [using that `UpdateKind` to properly handle captures](https://github.com/unoplatform/uno/blob/f6d17ecc8998257e09e1b37921370a8f7086a9dc/src/Uno.UI/UI/Xaml/Internal/RootVisual.cs#L176) so if you just ignore the property on older version of Android, you will have some weird results. What I would suggest, for older version of Android, is to use to keep the last pointer events and use the [`SetUpdateKindFromPrevious` extensions method](https://github.com/unoplatform/uno/blob/master/src/Uno.UWP/UI/Input/PointerPointPropertiesExtensions.cs#L14) that will actually compute that property based on previous state.

**Comment by queequac** [comment]
@dr1rrb To be honest, I'm kind of lost here. Where do I get the previous state to call this extension, etc.?
My gut tells me that you would fix this quite quickly, while I lack detailed knowledge about the current implementation of pointer events. :(

**Comment by jeromelaban** [comment]
@queequac we're not in the position of being able to fix this issue yet given our current priorities. 

You can give a try at your changes using our [contributing docs](https://platform.uno/docs/articles/uno-development/debugging-uno-ui.html#debugging-uno-in-another-application) using a custom Uno build, and let us know if you find something which may not work properly when using the change from this PR.

**Comment by jeromelaban** [comment]
> Maybe I can experiment also a little on the issue, but unfortunately I am not yet clear where to get the requiered things for being able to use that new extension to handle updatekinds properly.

One thing you can try is see how the app behaves on your device with the changes that are currently in this PR. You can try with the binaries found in the uitest-android-build artifact, from [this PR build](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=46004&view=artifacts&pathAsName=false&type=publishedArtifacts) :
image

**Comment by queequac** [comment]
I have to confess, I do not really understand how to make use of these artifacts.🙈
There are nuget packages inside, which allow to build the app, but it cannot be deployed becuse of missing uno tasks. There are assemblies which cannot be referenced because they are incompatible... and I have no clue how I'd use the APK to get something running.
Is there some guidance?

**Comment by jeromelaban** [comment]
You can use the APK to run on your device and click around to see if the behavior looks correct. You should see something that looks like this app: https://uno-ui-samplesapp-staging.azurewebsites.net/

As for the packages, you can find the nuget packages in the folder "vslatest" you can install locally:
image

**Comment by queequac** [comment]
I don't have a device with Android 5.1, just the telemetry from the play store that shows crashes on SDK 21 und 22. I was not able to deploy the sample app to the emulator, it always installed but no app appeared in the app list.
Anyway, I finally got the nuget packages working (no clue why it has thrown an error w.r.t some task)... and voila, it works again witht he guarding above. I can execute buttons, open panes, interact with hyperlinks and drop downs... no issues under Android 5.0 anymore. And I can ensure my app uses the build artifacts, since I show the versions of referenced open source libraries in app-about along with the license information. ;-)

Long story short: Would be very happy if that simple guarding would make it back into the main branch asap. 🤗

**Comment by queequac** [comment]
@jeromelaban @dr1rrb Any chance to get this merged? The nuget packages from that built seem to work fine with Android 5.0, I can click and navigate through my app. The entire UI and all controls are operable again.
Even if there would be some corner case where the event system might behave in an unexpected manner, this would be far better than an immediate crash on the very first touch.

**Comment by queequac** [comment]
What a pity this didn't make it into the 4.3 backport. :(
